sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -755,6 +755,40 @@ private static String buildColumnTypeString(ai.rapids.cudf.ColumnView view) {
     return type.toString();
   }
 
+  private static String buildColumnTypeString(DataType sparkType) {
+    DType dtype = toRapidsOrNull(sparkType);
+    if (dtype != null) {
+      return dtype.toString();
+    }
+    StringBuilder sb = new StringBuilder();
+    if (sparkType instanceof ArrayType) {
+      ArrayType arrayType = (ArrayType) sparkType;
+      sb.append("LIST(");
+      sb.append(buildColumnTypeString(arrayType.elementType()));
+      sb.append(")");
+    } else if (sparkType instanceof MapType) {
+      MapType mapType = (MapType) sparkType;
+      sb.append("LIST(STRUCT(");
+      sb.append(buildColumnTypeString(mapType.keyType()));
+      sb.append(",");
+      sb.append(buildColumnTypeString(mapType.valueType()));
+      sb.append("))");
+    } else if (sparkType instanceof StructType) {
+      StructType structType = (StructType) sparkType;
+      sb.append("STRUCT(");
+      for (int i = 0; i < structType.size(); i++) {
+        if (i != 0) {
+          sb.append(",");
+        }
+        sb.append(buildColumnTypeString(structType.apply(i).dataType()));
+      }
+      sb.append(")");
```

Comment:
can delegate to Scala
```suggestion
      sb.append(structType.iterator().map(sf -> buildColumnTypeString(sf.dataType()))
              .mkString("STRUCT(", ", ", ")"));
```